tests/Fixer/Whitespace/BlankLineBeforeControlStatementFixerTest.php

```diff
@@ -0,0 +1,1085 @@
+
+    /**
+     * @return array
+     */
+    public function providerFixWithRequire()
+    {
+        return [
+            [
+'<?php
+require "foo.php";',
+            ],
+            [
+'<?php
+$foo = $bar;
+
+require "foo.php";',
+'<?php
+$foo = $bar;
+require "foo.php";',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerFixWithRequireOnce
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithRequireOnce($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'statements' => ['require_once'],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithRequireOnce()
+    {
+        return [
+            [
+'<?php
+require_once "foo.php";',
+            ],
+            [
+'<?php
+$foo = $bar;
+
+require_once "foo.php";',
+'<?php
+$foo = $bar;
+require_once "foo.php";',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerFixWithReturn
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithReturn($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'statements' => ['return'],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithReturn()
+    {
+        return [
+            [
+'
+$a = $a;
+return $a;',
+            ],
+            [
+'<?php
+$a = $a;
+
+return $a;',
+'<?php
+$a = $a; return $a;',
+            ],
+            [
+'<?php
+$b = $b;
+
+return $b;',
+'<?php
+$b = $b;return $b;',
+            ],
+            [
+'<?php
+$c = $c;
+
+return $c;',
+'<?php
+$c = $c;
+return $c;',
+            ],
+            [
+'<?php
+$d = $d;
+
+return $d;',
+'<?php
+$d = $d;
+return $d;',
+            ],
+            [
+'<?php
+if (true) {
+    return 1;
+}',
+            ],
+            [
+'<?php
+if (true)
+    return 1;',
+            ],
+            [
+'<?php
+if (true) {
+    return 1;
+} else {
+    return 2;
+}',
+            ],
+            [
+'<?php
+if (true)
+    return 1;
+else
+    return 2;',
+            ],
+            [
+'<?php
+if (true) {
+    return 1;
+} elseif (false) {
+    return 2;
+}',
+            ],
+            [
+'<?php
+if (true)
+    return 1;
+elseif (false)
+    return 2;',
+            ],
+            [
+'<?php
+throw new Exception("return true;");',
+            ],
+            [
+'<?php
+function foo()
+{
+    // comment
+    return "foo";
+}',
+            ],
+            [
+'<?php
+function foo()
+{
+    // comment
+
+    return "bar";
+}',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerFixWithReturnAndMessyWhitespaces
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithReturnAndMessyWhitespaces($expected, $input = null)
```

Comment:
Instead of using specific test cases for different indent and line ending, I would reuse `providerFixWithReturn()` data provider and `str_replace("\n", "\r\n", ...)` on input and expected codes before calling `$this->doTest(...)`. I would also do that for all other keywords.